runtime/memprof.c

```diff
@@ -464,17 +520,20 @@ static void flush_deleted(void)
       j++;
     }
     i++;
-    if (trackst.young == i) trackst.young = j;
-    if (trackst.callback == i) trackst.callback = j;
+    if (trackst.young_idx == i) trackst.young_idx = j;
+    if (trackst.promote_dealloc_callback_idx == i)
+      trackst.promote_dealloc_callback_idx = j;
   }
```

Comment:
I don't see how the next_alloc_callback_idx fields are maintained when everything shuffles around here

Reply:
Indeed, this is completely bogus. This is not because of `next_alloc_callback_idx`, but rather because of the idx_ptr pointers which are not updated accordingly.